File: URI/tle/uri2878.py
class Corte:
    def __init__(self, ini, fim):
        self.ini = ini
        self.fim = fim

    def intersect(self, outro_corte):
        if( self.ini < outro_corte.ini and self.fim > outro_corte.fim ) or ( self.ini > outro_corte.ini and self.fim < outro_corte.fim ):
            return True
        return False


linha = input().split()
x = int(linha[0])
y = int(linha[1])

linha = input().split()
h = int(linha[0])
v = int(linha[1])

horizontais = []
for i in range(h):
    linha = input().split()
    horizontais.append( Corte(int(linha[0]), int(linha[1])) )

verticais = []
for i in range(v):
    linha = input().split()
    verticais.append( Corte(int(linha[0]), int(linha[1])) )

horizontais.sort(key=lambda corte: corte.ini)
verticais.sort(key=lambda corte: corte.ini)

res = (h+1)*(v+1)

for i in range(h-1):
    for j in range(i+1, h):
        if( horizontais[i].intersect(horizontais[j]) ):
            res += 1

for i in range(v-1):
    for j in range(i+1, v):
        if( verticais[i].intersect(verticais[j]) ):
            res += 1

print(res)

File: URI/tle/uri2878.java
import java.io.*;
import java.util.*;
class Main {
    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] linha = br.readLine().split(" ");
        int x = Integer.parseInt(linha[0]);
        int y = Integer.parseInt(linha[1]);

        linha = br.readLine().split(" ");
        int h = Integer.parseInt(linha[0]);
        int v = Integer.parseInt(linha[1]);

        Corte[] horiz = new Corte[h];
        for(int i = 0; i < h; i++) {
            linha = br.readLine().split(" ");
            int ini = Integer.parseInt(linha[0]);
            int fim = Integer.parseInt(linha[1]);

            horiz[i] = new Corte(ini,fim);
        }

        Corte[] verti = new Corte[v];
        for(int i = 0; i < v; i++) {
            linha = br.readLine().split(" ");
            int ini = Integer.parseInt(linha[0]);
            int fim = Integer.parseInt(linha[1]);

            verti[i] = new Corte(ini,fim);
        }

        Arrays.sort(horiz);
        Arrays.sort(verti);

        int res = (h+1)*(v+1);

        TreeSet<Integer> ss = new TreeSet<Integer>();
        for(Corte c: horiz) {
            res += ss.tailSet(c.fim).size();
            ss.add(c.fim);
        }

        ss = new TreeSet<Integer>();
        for(Corte c: verti) {
            res += ss.tailSet(c.fim).size();
            ss.add(c.fim);
        }

        System.out.println(res);

    }
}

class Corte implements Comparable {
    Integer ini, fim;
    public Corte(int x, int y) {
        this.ini = x;
        this.fim = y;
    }
    public int compareTo(Object obj) {
        Corte outro = (Corte)obj;
        return ini.compareTo(outro.ini);
    }
}